sqswatcher/sqswatcher.py

```diff
@@ -99,15 +104,19 @@ def _setup_queue(region, queue_name, proxy_config):
     :param proxy_config: proxy configuration
     :return: the Queue object
     """
-    log.debug("running _setup_queue")
-
+    log.debug("Getting SQS queue '%s'", queue_name)
     sqs = boto3.resource("sqs", region_name=region, config=proxy_config)
+    try:
+        queue = sqs.get_queue_by_name(QueueName=queue_name)
+    except ClientError as e:
+        log.critical("Unable to get the SQS queue '%s'. Failed with exception: %s", queue_name, e)
+        raise
 
-    _queue = sqs.get_queue_by_name(QueueName=queue_name)
-    return _queue
+    return queue
 
 
-def _setup_ddb_table(region, table_name, proxy_config):
+@retry(stop_max_attempt_number=3, wait_fixed=5000)
```

Comment:
you probably don't want to retry on CriticalErrors. This should work:
```
@retry(
...
retry_on_exception=lambda exception: not isinstance(exception, CriticalError)
)
```

Reply:
fixed